File: typescript/libs/workspace/domain/system/src/lib/services/IWorkspaceServices.ts
import { ISessionServices } from "./ISessionServices";
import { IWorkspace } from "../IWorkspace";
import { IDerivationService } from "./derivation/IDerivationService";

export interface IWorkspaceServices {
  onInit(workspace: IWorkspace): void;

  createSessionServices(): ISessionServices;

  derivationService: IDerivationService;
}

File: dotnet/System/Workspace/Allors.Workspace.Domain/Services/IWorkspaceServices.cs
// <copyright file="IWorkspaceServices.cs" company="Allors bv">
// Copyright (c) Allors bv. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace
{
    using System;

    public interface IWorkspaceServices : IDisposable
    {
        void OnInit(IWorkspace workspace);

        ISessionServices CreateSessionServices();

        T Get<T>();
    }
}
